astropy/modeling/tabular.py

```diff
@@ -0,0 +1,193 @@
+
+"""
+
+from __future__ import (absolute_import, unicode_literals, division,
+                        print_function)
+import abc
+import numpy as np
+from .core import Model
+from ..utils import minversion
+
+try:
+    import scipy
+    from scipy.interpolate import interpn
+    has_scipy = True
+except ImportError:
+    has_scipy = False
+
+if has_scipy and not minversion(scipy, "0.14"):
+    has_scipy = False
+
+
+__all__ = ['tabular_model', 'Tabular1D', 'Tabular2D']
+
+__doctest_requires__ = {('Tabular', 'tabular_model'): ['scipy']}
+
+
+class Tabular(Model):
+    """
+    Returns an interpolated lookup table value.
+
+    Parameters
+    ----------
+    points : tuple of ndarray of float, with shapes (m1, ), ..., (mn, ), optional
+        The points defining the regular grid in n dimensions.
+    lookup_table : array_like, shape (m1, ..., mn, ...)
+        The data on a regular grid in n dimensions.
+    method : str, optional
+        The method of interpolation to perform. Supported are "linear" and
+        "nearest", and "splinef2d". "splinef2d" is only supported for
+        2-dimensional data. Default is "linear".
+    bounds_error : bool, optional
+        If True, when interpolated values are requested outside of the
+        domain of the input data, a ValueError is raised.
+        If False, then `fill_value` is used.
+    fill_value : float, optional
+        If provided, the value to use for points outside of the
+        interpolation domain. If None, values outside
+        the domain are extrapolated.  Extrapolation is not supported by method
+        "splinef2d".
+
+    Returns
+    -------
+    value : ndarray
+        Interpolated values at input coordinates.
+
+    Raises
+    ------
+    ImportError
+        Scipy is not installed.
+
+    Notes
+    -----
+    Uses ``scipy.interpolate.interpn``
+
+    """
+
+    linear = False
+    fittable = False
+
+    standard_broadcasting = False
+    outputs = ('y',)
+
+    lookup_table = abc.abstractproperty()
+
+    _is_dynamic = True
+
+    _id = 0
+
+    def __init__(self, points=None, lookup_table=None, method='linear',
+                 bounds_error=True, fill_value=np.nan, **kwargs):
+
+        super(Tabular, self).__init__(**kwargs)
+        if lookup_table is not None:
+            lookup_table = np.asarray(lookup_table)
+            if self.lookup_table.ndim != lookup_table.ndim:
+                raise ValueError("lookup_table should be an array with "
+                                 "{0} dimensions".format(self.lookup_table.ndim))
+            self.lookup_table = lookup_table
+        if points is None:
+            self._points = tuple(np.arange(x, dtype=np.float)
+                                 for x in self.lookup_table.shape)
+        else:
+            if self.lookup_table.ndim == 1:
+                self._points = (points,)
+            else:
+                self._points = points
```

Comment:
I think converting it to a format for `interpn` should not be exposed publicly as `self.points`. This is because someone might decide to subclass `Tabular` and define their own way of interpolation using a completely different API. It would be messy if they have to re-convert the points from this special kind of tuple back to normal arrays for their usage.

tl;dr -- Public-facing `points` should be generic. Points re-formatted for `interpn` should be private and not affect subclass overwritting `evaluate()`.